fix: various multibackend issues - WPB-20257

johnxnguyen opened the pull request:
<!--do not remove this marker, its needed to replace info when ticket title is updated -->
<!--jira-description-action-hidden-marker-start-->

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-20257" title="WPB-20257" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />WPB-20257</a>  [iOS] Check blacklist when loading user session
  </td></table>
  <br />
 

<!--jira-description-action-hidden-marker-end-->
<!--do not remove the jira markers to link tickets automatically -->

Also fixes: WPB-20258 WPB-20259

### Issue
This PR fixes a few issues with multibackend:
1. **The blacklist is now checked every time the session loads:** this prevents the case where the app is blacklisted, but relaunching the app would bypass the blacklist until the next time it's checked (every 6 hours).
2. **Unable to switch accounts on the blocker view controller:** this was because if the app was blacklisted we were ignoring any new app state changes, but this doesn't make sense anymore in multibackend environments. Additionally, it was possible to dismiss the account picker and be stuck.
3. **Sometimes when logging in and deleting a client it would get stuck:** and you'd be left in a broken state which would require a new install. This was due to a race condition where sometimes the user session was not yet available through a singleton and therefore there would be no use case to delete the client.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

johnxnguyen committed:
prevent account switcher being dismissed

johnxnguyen committed:
fix ui not changing when switching accounts

johnxnguyen committed:
fix not being about to delete a client after login

johnxnguyen committed:
check blacklist when loading session

johnxnguyen committed:
remove debug condition

johnxnguyen committed:
add developer overrides to make testing easier

johnxnguyen committed:
fix blacklist override

johnxnguyen committed:
fix compilation

johnxnguyen committed:
fix blacklisting in session loader

johnxnguyen committed:
tear down session before exiting

johnxnguyen committed:
fix typo

johnxnguyen committed:
revert flag

johnxnguyen committed:
fix typo

netbe reviewed (commented):
left some questions before approving

johnxnguyen committed:
Update wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession+Authentication.swift

Co-authored-by: François Benaiteau <netbe@users.noreply.github.com>

johnxnguyen committed:
make overrides available in extensions

# Conflicts:
#	WireFoundation/Sources/WireFoundation/Utilities/DeveloperOverrides.swift

johnxnguyen committed:
improve overrides form

johnxnguyen committed:
format

johnxnguyen committed:
format

johnxnguyen committed:
fix tests
